### ISSUE TITLE
[bug report] wrong implementation of gradient clip in WGAN?

### ISSUE THREAD

**Issue by KK666-AI** [znxlwm/pytorch-generative-model-collections#31]
Dear author,

Thanks for your friendly implementation of popular GANs. I am reading your implementation of [Wasserstein GANs](https://github.com/znxlwm/pytorch-generative-model-collections/blob/master/WGAN.py), and notice that the gradient clip operation is placed after optimizer.step as seen in this [line](https://github.com/znxlwm/pytorch-generative-model-collections/blob/0d183bb5ea2fbe069e1c6806c4a9a1fd8e81656f/WGAN.py#L155).
Typically, in pytorch, the gradient operation is placed before the optimizer step.

pls have a check.

**Comment by Wuziyi616** [comment]
As you can see in this line `for p in self.D.parameters():` this is not performing gradient clipping but params value clipping
